x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/fielddata/ShapeValues.java

```diff
@@ -117,6 +117,13 @@ public BoundingBox boundingBox() {
             return boundingBox;
         }
 
+        /**
+         * Visit the underlying tree structure using the provided {@link TriangleTreeVisitor}
+         */
+        public void visit(TriangleTreeVisitor visitor) throws IOException {
```

Comment:
I presume this is the new method you plan to use?

Reply:
yes, I think it makes sense to have such a method instead of adding a new method whenever you want to add new functionality.